sdk/core/core/inc/az_span.h

```diff
@@ -260,6 +260,32 @@ AZ_NODISCARD az_result az_span_to_uint64(az_span span, uint64_t* out_number);
  */
 AZ_NODISCARD az_result az_span_to_uint32(az_span span, uint32_t* out_number);
 
+/**
+ * @brief az_span_find searches for `target` in `source`, returning an #az_span within `source` if it finds it.
+ *
+ * @param[in] source The #az_span with the content to be searched on.
+ * @param[in] target The #az_span containing the token to be searched in `source`.
+ * @param[out] out_span The #az_span pointing to the first occurrence of `target` in `source`, if it is found.
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK if `target` is found in `source`
+ *          #AZ_ERROR_ITEM_NOT_FOUND if `target` is not found in `source`
+ * is found within the span.
+ */
+AZ_NODISCARD az_result az_span_find(az_span source, az_span target, az_span* out_span);
+
+/**
+ * @brief az_span_token is a string tokenizer for az_span.
+ *
+ * @param[in] source The #az_span with the content to be searched on. It must be a non-empty #az_span.
+ * @param[in] delimiter The #az_span containing the delimiter to "split" `source` into tokens.  It must be a non-empty #az_span.
+ * @param[out] out_remainder The #az_span pointing to the remaining bytes in `source`, starting after the occurrence of `delimiter`.
+ *                        If the position after `delimiter` is the end of `source`, `out_remainder` is set to a NULL/empty #az_span.
```

Comment:
nit: Remove `NULL`. Just state `empty #az_span`. A NULL span is a bit confusing since it's a struct.

Also, why is this line so deeply indented?
```suggestion
 *                        If the position after `delimiter` is the end of `source`, `out_remainder` is set to an empty #az_span.
```

Reply:
`AZ_SPAN_NULL` is maybe what to have instead. Or “empty”